app/bundles/EmailBundle/EventListener/MatchFilterForLeadTrait.php

```diff
@@ -151,23 +151,36 @@ protected function matchFilterForLead(array $filter, array $lead)
                     break;
                 case 'in':
                     $leadValMatched = false;
-                    foreach ($leadVal as $k => $v) {
-                        if (in_array($v, $filterVal)) {
+                    if (getType($leadVal) == 'array') {
```

Comment:
Please use [is_array](http://php.net/manual/en/function.is-array.php) instead. (plus once more bellow)